### PR TITLE
create PodChaos by API

### PR DESCRIPTION
**Pull request by cwen0**
Signed-off-by: cwen0 <cwenyin0@gmail.com>

### What problem does this PR solve?  

#485 

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

JSON data: 

```
{
  "name": "pod-chaos-1",
  "namespace": "default",
  "scope": {
    "namespace_selectors": [
      "default"
    ],
    "label_selectors": {
      "app": "web-show"
    },
    "mode": "one"
  },
  "target": {
    "kind": "PodChaos",
    "pod_chaos": {
      "action": "pod-kill"
    }
  }, 
  "scheduler": {
    "cron": "@every 30s"
  }
}
``` 

result: 
image

### CODE REVIEW

File: pkg/apiserver/experiment/experiment.go [hunk 1]
```diff
@@ -70,8 +75,139 @@ func (s *Service) getExperimentDetail(c *gin.Context) {}
...
+
+func (s *ScopeInfo) parseSelector() v1alpha1.SelectorSpec {
+	selector := v1alpha1.SelectorSpec{}
+
+	for _, ns := range s.NamespaceSelectors {
+		selector.Namespaces = append(selector.Namespaces, ns)
+	}
+
+	for key, val := range s.LabelSelectors {
+		selector.LabelSelectors = make(map[string]string)
```
Comment (yeya24) [inline pkg/apiserver/experiment/experiment.go:106]: This feels wrong. Why initializing this map every iteration?

File: pkg/apiserver/experiment/experiment.go [hunk 2]
```diff
@@ -70,8 +75,139 @@ func (s *Service) getExperimentDetail(c *gin.Context) {}
...
+		selector.LabelSelectors[key] = val
+	}
+
+	for key, val := range s.AnnotationSelectors {
+		selector.AnnotationSelectors = make(map[string]string)
```
Comment (yeya24) [inline pkg/apiserver/experiment/experiment.go:111]: ditto

File: pkg/apiserver/experiment/experiment.go [hunk 3]
```diff
@@ -70,8 +75,139 @@ func (s *Service) getExperimentDetail(c *gin.Context) {}
...
+		selector.AnnotationSelectors[key] = val
+	}
+
+	for key, val := range s.FieldSelectors {
+		selector.FieldSelectors = make(map[string]string)
```
Comment (yeya24) [inline pkg/apiserver/experiment/experiment.go:116]: ditto

File: pkg/apiserver/experiment/experiment.go (hunk 1 above)
Comment (cwen0) [reply pkg/apiserver/experiment/experiment.go:106]: Got

File: pkg/apiserver/experiment/experiment.go [hunk 4]
```diff
@@ -70,8 +75,139 @@ func (s *Service) getExperimentDetail(c *gin.Context) {}
 // TODO: need to be implemented
 func (s *Service) deleteExperiment(c *gin.Context) {}
 
+// ExperimentInfo defines a form data of Experiment from API.
+type ExperimentInfo struct {
+	Name      string        `json:"name" binding:"required,NameValid"`
+	Namespace string        `json:"namespace" binding:"required,NameValid"`
+	Scope     ScopeInfo     `json:"scope"`
```
Comment (Yisaer) [inline pkg/apiserver/experiment/experiment.go:82]: What about add `Labels` and `Annotations` to let user fulfill the target podchao's labels and annotations.

File: pkg/apiserver/experiment/experiment.go [hunk 5]
```diff
@@ -70,8 +75,139 @@ func (s *Service) getExperimentDetail(c *gin.Context) {}
...
+	IOChaos      IOChaosInfo
+	KernelChaos  KernelChaosInfo
+	TimeChaos    TimeChaosInfo
+	StressChaos  StressChaosInfo
+}
+
+// SchedulerInfo defines the scheduler information.
+type SchedulerInfo struct {
+	Cron string `json:"cron" binding:"CronValid"`
+}
```
Comment (Yisaer) [inline pkg/apiserver/experiment/experiment.go:141]: Where is the `Duration`? I guess people may use `duration` for `pod-failure` chaos.

File: doc/docs.go [hunk 6]
```diff
@@ -1,6 +1,6 @@
 // GENERATED BY THE COMMAND ABOVE; DO NOT EDIT
 // This file was generated by swaggo/swag at
-// 2020-05-20 14:44:28.92534 -0400 EDT m=+0.517600077
+// 2020-05-21 11:13:08.835806 +0800 CST m=+1.102151256
```
Comment (yeya24) [inline doc/docs.go:3]: Does this time update even if no swagger field is changed? This seems annoying...

Comment (cwen0) [reply doc/docs.go:3]: Yes, If we don't build swagger UI, this won't be updated.

**Pull request merged**